cognite/client/_api/simulators/models.py

```diff
@@ -0,0 +1,437 @@
+        """Iterate over simulator simulator model revisions
+
+        Fetches simulator model revisions as they are iterated over, so you keep a limited number of simulator model revisions in memory.
+
+        Args:
+            chunk_size (int | None): Number of simulator model revisions to return in each chunk. Defaults to yielding one simulator model revision a time.
+            filter (SimulatorModelRevisionsFilter | None): Filter to apply on the model revisions list.
+            limit (int | None): Maximum number of simulator model revisions to return. Defaults to return all items.
+
+        Returns:
+            Iterator[SimulatorModelRevision] | Iterator[SimulatorModelRevisionList]: yields Simulator one by one if chunk is not specified, else SimulatorList objects.
+        """
+        return self._list_generator(
+            list_cls=SimulatorModelRevisionList,
+            resource_cls=SimulatorModelRevision,
+            method="POST",
+            filter=filter.dump() if isinstance(filter, CogniteFilter) else filter,
+            chunk_size=chunk_size,
+            limit=limit,
+        )
+
+    @overload
+    def create(
+        self, revisions: Sequence[SimulatorModelRevision]
+    ) -> SimulatorModelRevision | SimulatorModelRevisionList: ...
+
+    @overload
+    def create(
+        self, revisions: SimulatorModelRevision | SimulatorModelRevisionWrite
+    ) -> SimulatorModelRevision | SimulatorModelRevisionList: ...
+
+    def create(
+        self,
+        revisions: SimulatorModelRevision
+        | SimulatorModelRevisionWrite
+        | Sequence[SimulatorModelRevision]
+        | Sequence[SimulatorModelRevisionWrite],
+    ) -> SimulatorModelRevision | SimulatorModelRevisionList:
+        """`Create one or more simulator model revisions. <https://api-docs.cognite.com/20230101-beta/tag/Simulator-Models/operation/create_simulator_model_revision_simulators_models_revisions_post>`_
+        You can create an arbitrary number of simulator model revisions, and the SDK will split the request into multiple requests.
+        Args:
+            revisions (SimulatorModelRevision | SimulatorModelRevisionWrite | Sequence[SimulatorModelRevision] | Sequence[SimulatorModelRevisionWrite]): Simulator model or list of Simulator models to create.
+        Returns:
+            SimulatorModelRevision | SimulatorModelRevisionList: Created simulator model(s)
+        Examples:
+            Create new simulator models:
+                >>> from cognite.client import CogniteClient
+                >>> from cognite.client.data_classes import SimulatorModelRevision
+                >>> client = CogniteClient()
+                >>> models = [SimulatorModelRevision(external_id="model1"), SimulatorModelRevision(external_id="model2")]
+                >>> res = client.simulators.models.create_revision(models)
+        """
+        assert_type(revisions, "simulator_model_revision", [SimulatorModelRevisionCore, SimulatorModelRevisionWrite])
```

Comment:
Two questions:
1. Why do you specify both SimulatorModelRevisionCore and its subclass SimulatorModelRevisionWrite?
2. The create method says "...or list of Simulator models to create". The current code would block that I think

Reply:
1 and 2 - I initially didn’t fully grasp how the core and write classes were connected, but I’ve (sort of 🙈 ) figured it out now. Sorry about that!